### PR TITLE
Added PKCE example for PHP

### PR DESCRIPTION
**Pull request by KMChappell**
This PR adds a "get-pkce-access-token" script that shows how to get an access token or refresh an access token when using a public client type with PHP.

- tested both getting and refreshing tokens
- tested each other sample after using the get token script to obtain a new access token

### CODE REVIEW

File: php/get-pkce-access-token [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    $clientId = rtrim(fgets(STDIN), PHP_EOL);
+    $redirectUri = "urn:ietf:wg:oauth:2.0:oob";
+
+    // Generate the code challenge using the OS / cryptographic random function
+    $verifierBytes = random_bytes(64);
+    $codeVerifier = rtrim(strtr(base64_encode($verifierBytes), "+/", "-_"), "=");
+
+    // Very important, "raw_output" must be set to true or the challenge
+    // will not match the verifier.
+    $challengeBytes = hash("sha256", $codeVerifier, true);
```
Comment (cknave) [inline php/get-pkce-access-token]: To use the hash extension we should have `"ext-hash": "*"` to the requirements in composer.json

File: php/get-pkce-access-token [hunk 2]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env php
```
Comment (cknave) [inline php/get-pkce-access-token]: This file needs execute permission

File: php/get-pkce-access-token (hunk 1 above)
Comment (KMChappell) [reply php/get-pkce-access-token]: Whoops, I forgot to commit composer.json. Added it in my recent commit.

File: php/get-pkce-access-token (hunk 2 above)
Comment (KMChappell) [reply php/get-pkce-access-token]: `ls -l` now matches the other file permissions, so this should be fixed in my most recent commit.

**Pull request merged**